apps/extension/client/src/utils.ts

```diff
@@ -0,0 +1,13 @@
+export const debounce = <T extends (...args: any[]) => ReturnType<T>>(
+  callback: T,
+  timeout: number
+): ((...args: Parameters<T>) => void) => {
+  let timer: ReturnType<typeof setTimeout>;
```

Comment:
```suggestion
  let timer: number;
```
`setTimeout` always returns a number, so it felt weird to use return type.

Reply:
It doesn't return a number in node JS :D It might be easier to grab the correct type here using `ReturnType` like it has been done than to try to remember how to reach for that type manually (it's `NodeJS.Timer` in node)

Reply:
TIL!